src/index.js

```diff
@@ -41,12 +46,52 @@ export const RCComponent = ({
   );
 
   useEffect(() => {
-    const cookiesPresent = Cookies.get('rc_token') && Cookies.get('rc_uid');
+    const cookiesPresent =
+      Cookies.get(RC_USER_TOKEN_COOKIE) && Cookies.get(RC_USER_ID_COOKIE);
     if (cookiesPresent) {
       setIsUserAuthenticated(true);
     }
   }, []);
 
+  const authenticatedUserUsername = useUserStore((state) => state.username);
```

Comment:
I was thinking about this part, and as you can see that this has bloated our index.js file with all the useUserStore hooks -> can we shift it to a generic custom hook? Something like -> `useUser`

The idea is to reuse the functionality in logout too where we destroy the local stores.
Refer:
https://github.com/RocketChat/EmbeddedChat/blob/a8d8ad31e65c752b31fc87f62ee33b385cb2d234/src/components/ChatHeader/ChatHeader.js#L35

We could handle all these logic there and keep it simple inside index.js file

Reply:
I got your point you want to refractor that code little bit but i am not so good with react and react hooks 😅i am more of a backend guy 

Can you explain me this hook thing bit more or point me to some resources or reference please that will help alot

Reply:
I made the useUser hook and also used that in all files where useUserstore was used before and I also added Eslint comments `/* eslint-disable no-shadow */ `  ,  ` /* eslint-disable no-nested-ternary */
` 
 to fix the error in Chatbody , Chatinput ,useUser hook

Reply:
Please have a look @sidmohanty11  and let me know if any changes required

Reply:
I researched about it for you Linting is the process of running a program that analyzes your code for potential errors or issues. It can help you find syntax errors, undefined variables, and other issues that can cause problems when you try to run your code. Linting can be a useful tool for catching errors early on and improving the overall quality of your code.

Linting forces us to Coding conventions are guidelines that recommend how you should format and organize your code. These conventions can include things like how you should name variables, where you should put whitespace, and how you should structure your code. Coding conventions are used to make code more consistent and easier to read and if some file is already written without following this convention we could add a comment telling eslint to ignore conventions for this file or this line

Reply:
What?

Reply:
i mean i know why we are using this comments because of linting

Reply:
I know its because of linting :( not sure what you mean

Reply:
no worries i removed the redundant code and checked everything running fine